### PR TITLE
Add buddy system

### LINKED ISSUE #2: Setup a buddy system to help onboard new developers that makes use of e.g. pair programming.

### PR DESCRIPTION
**Pull request by diegodelemos**
Closes #2

### CODE REVIEW

File: projects/buddy-system.md [hunk 1]
```diff
@@ -0,0 +1,98 @@
...
+The buddy must meet the newcomer’s supervisor in order to get an overview of her/his project and discuss the first tasks:
+
+* Schedule when supervisor will introduce buddy to the newcomer
+* Define tasks on Invenio
+  * One of the following small tasks: read up relevant Invenio documentation, rehearse the presentations, test tutorials, check if any part of the material is outdated (~1 hr)
+  * A medium sized task (~1 day) suitable for pair-programming
+  * A first larger service task in which the supervisor will help
+* Agree on who is going to run the bootcamp following `inveniosoftware/training` repository.
+
+### First day
```
Comment (jbenito3) [inline projects/buddy-system.md]: I must confess that what you are proposing as First Day it is exactly what I do :) (and I would bet that it is what every supervisor will do :) ).

File: projects/buddy-system.md [hunk 2]
```diff
@@ -0,0 +1,98 @@
...
+* Guided tutorial: Developing for Invenio, including cookiecutter (+seeing all the files), tiny task, running tests, installing inside Invenio
+* Pair-programming: Solve first small task
+* Tutorial: Submitting a pull request
+* Presentation: Overview of contribution and style guide
+* Pair-programming: Solve medium sized task
+* Presentation: Invenio general architecture overview
+* Presentation: Module overview
+* Plan for pair-programming on a service task
+
+**Remember**, newcomers also need “free” time for setting up their environment, office space etc ...
```
Comment (jbenito3) [inline projects/buddy-system.md]: Shall we say something about being flexible with newcomers also depending on their expertise? There will be some (less experienced) that require more assistant, other ones (more experienced perhaps) that will appreciate the introduction and will prefer to continue on their own till they hit a blocker. WDYT?
It is difficult to express in words, but what I mean is that we should be able to read the newcomer needs, not force her/him to be trained but to understand what level of assistance she/he needs. WDYT @diegodelemos ?

Comment (diegodelemos) [inline projects/buddy-system.md]: fully agree, I will try to reformulate it

File: projects/buddy-system.md [hunk 3]
```diff
@@ -0,0 +1,98 @@
...
+
+* Newcomer should already feel comfortable enough to ask for help whenever needed
+* Be accessible for newcomer
+* Check that pull requests and interaction with the section is appropriate (i.e. according to CERN and Invenio code of conduct)
+* Proactively ask for feedback
+
+### 8th week
+
+* Final feedback meeting with buddy (always take note about how to improve the system)
+* Retrospective meeting about last two months with buddy system maintainer
```
Comment (jbenito3) [inline projects/buddy-system.md]: I am missing a last point "What happens next?" :) I mean, after the the 2 months the 'official' assistance is over but of course, they can still talk and the newcomer can still ask questions, nope? hehehe :)

File: projects/buddy-system.md (hunk 1 above)
Comment (diegodelemos) [inline projects/buddy-system.md]: Completely agree. Just about social events (like evening events) and official events (like Invenio Dev Forum), we leave this also to section leader and supervisor? Or perhaps creating a new section about it like social, or even adding them as a single line in the general duties (Something like, `Make sure that newcomer doesn't miss our social and official events`).

Comment (jbenito3) [inline projects/buddy-system.md]: I think you are right, you can speak about the social and official events. Even if we mention it twice is not a big deal. It is better to make sure they know what's going on that forgetting to do so (I or some supervisor can forget some of this info) :)

**Comment by diegodelemos** [comment]
updated

File: projects/buddy-system.md [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+* Proactively ask for feedback
+
+#### 8th week
+
+* Final feedback meeting with buddy (always take note about how to improve the system)
+* Retrospective meeting about last two months with buddy system maintainer
+
+## What happens next
+
+Of course after the two months the 'official assistance' is over but, it doesn't finish here, you will still be in contact with the not that newcomer anymore and he can still ask questions or even help you back!
```
Comment (jbenito3) [inline projects/buddy-system.md]: "with the not that" :D

**Pull request merged**